test/bats/test.bats

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bats
+
+load helper
+
+BATS_TESTS_DIR=test/bats/tests
+WAIT_TIME=60
+SLEEP_TIME=1
+
+@test "install Opa-Istio" {
+  run kubectl apply -f quick_start.yaml
+  assert_success
+}
+
+@test "label default namespace for Istio Proxy and OPA-Istio sidecar injection" {
+  run kubectl label namespace default opa-istio-injection="enabled"
+  assert_success
+
+  run kubectl label namespace default istio-injection="enabled"
+  assert_success
+}
+
+@test "deploy Bookinfo app" {
+  run kubectl apply -f https://raw.githubusercontent.com/istio/istio/master/samples/bookinfo/platform/kube/bookinfo.yaml
+  assert_success
+
+  cmd="kubectl wait --for=condition=available --timeout=60s --all deployment"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+  assert_success
+}
+
+@test "make Bookinfo app accessible from outside the cluster" {
+  run kubectl apply -f https://raw.githubusercontent.com/istio/istio/master/samples/bookinfo/networking/bookinfo-gateway.yaml
+  assert_success
+}
+
+@test "alice can access /productpage" {
+  host="$(docker_ip 'kind-control-plane')"
+  port="$(ingress_port)"
+
+  echo "$host"
+  echo "$port"
+
+  result="$(docker exec kind-control-plane curl --connect-timeout 30 --max-time 60 --retry 10 --retry-delay 0 --retry-max-time 600 --retry-connrefused --user alice:password -s -o /dev/null -w "%{http_code}" http://"$host":"$port"/productpage)"
```

Comment:
It might make sense to have a variable in the helpers (or a function) that can do the curl's with all the like timeouts, retries, etc defined. Would simplify these lines quite a bit to make it easier to see what is being tested for.

At a minimum I'd request that the line get broken up with `\` so there isn't any horizontal scrolling needed on narrow views 😄

Reply:
Broke up the lines to assist visibility.